WordPress/src/main/java/org/wordpress/android/datasets/ReaderTagTable.java

```diff
@@ -363,4 +372,31 @@ private static int minutesSinceLastUpdate(ReaderTag tag) {
         Date dtNow = new Date();
         return DateTimeUtils.minutesBetween(dtUpdated, dtNow);
     }
+
+    public static void deleteRecommendedTagsAddedAsFollowedTags() {
+        // Delete recommended tags which were saved as followed tags for logged out user
+        // before we allowed following tags using interests picker
+        if (!AppPrefs.getReaderRecommendedTagsDeletedForLoggedOutUser()) {
+            deleteTagsAndPostsWithTags(getFollowedTags());
+            AppPrefs.setReaderRecommendedTagsDeletedForLoggedOutUser(true);
+        }
+    }
+
+    public static void deleteTagsAndPostsWithTags(ReaderTagList tagList) {
+        if (tagList == null || tagList.size() == 0) {
+            return;
+        }
+        deleteTags(tagList);
+
+        SQLiteDatabase db = ReaderDatabase.getWritableDb();
+        db.beginTransaction();
+        try {
+            for (ReaderTag tag : tagList) {
+                ReaderPostTable.deletePostsWithTag(tag);
```

Comment:
Do we really need to delete the posts here? It feels a bit weird to access one table from another. I think "dangling" posts will get deleted automatically [here](https://github.com/wordpress-mobile/WordPress-Android/blob/f7760b32bd2cc702618858e1b941c2d92089505d/WordPress/src/main/java/org/wordpress/android/datasets/ReaderPostTable.java#L215). Wdyt?

Reply:
It was an existing method that I moved from `ReaderUpdateLogic` to the `ReaderTagTable`.  I suppose posts deletion can be handled at the line you linked.

Reply:
If we remove the tags in the migration script I believe we won't need this method, right?

Reply:
Removed this method and used `deleteTags() `directly when local tags are synced with the server in in 5c6f8e3.